hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -88,7 +91,18 @@ public static SparkConf registerClasses(SparkConf conf) {
 
   @Override
   protected HoodieIndex<T, JavaRDD<HoodieRecord<T>>, JavaRDD<HoodieKey>, JavaRDD<WriteStatus>> createIndex(HoodieWriteConfig writeConfig) {
-    return SparkHoodieIndex.createIndex(config);
+    String persistIndexType = null;
+    try {
+      persistIndexType = this.createMetaClient(true).getTableConfig().getProperties().getProperty(HoodieIndexConfig.INDEX_TYPE_PROP);
```

Comment:
should we create a new metaClient. Can we see if we can pass in metaClient from the caller?

Reply:
can not get a general metaClient in caller  AbstractHoodieWriteClient. otherwise we can move the createMetaClient to AbstractHoodieWriteClient , then introduct metaclient to createIndex, and modify the abstract method from  createIndex(HoodieWriteConfig writeConfig)  to createIndex(HoodieWriteConfig writeConfig, HoodieTableMetaClient metaClient).
@vinothchandar  what do you think about?

Reply:
MetaClient does not really depend on any engine specific context. So we could attempt that. Creating metaclient reloads the timeline again, which can affect the `snapshot` that the writer sees. Right now, we just do it once before we begin any write operation and its conceptually simple to understand the rest of the code. Love to keep it that way. @lw309637554 I can take a stab at it as well, if you feel its a bit complex. do let me know!

Reply:
thanks, we can reuse metaclient in AbstractHoodieWriteClient , for the reload activetimeline , can use  metaClient.reloadActiveTimeline();